### DISCUSSION TITLE
Publicode ou publicodes

### DISCUSSION THREAD

**Discussion by fm89** [publicodes/publicodes#44] [General]
La page https://publi.codes/ semble se référer au langage à la fois au singulier "Publicode est un langage ... ", "Publicode est adapté ..." et au pluriel "mon-entreprise.fr utilise publicodes ...", "futur.eco utilise publicodes ..." 

Est-ce souhaité ou un reste du passé à aligner vers votre choix final ?

**Comment by mquandalle** [comment]
Ce n'est pas un reste du passé, nous ne nous sommes jamais vraiment arrêtés sur une convention. Pour ma part j'ai souvent utilisé Publicode au singulier car je trouve plus intuitif d'utiliser le singulier pour parler d'un langage (On utilise Javascript pas Javascripts même s'il y a plusieurs "scripts"). Mais effectivement le nom de domaine est en `.codes` ce qui donne un argument en faveur de l'utilisation du pluriel.

Pour ma part je garde une préférence pour le singulier. Utilisons cette issue pour arbitrer ce point puis nous pourrons mettre à jour le site.

**Comment by johangirod** [comment]
Je suis également favorable au singulier. J'avais commencé à ajouter des "s" de partout pour être raccord avec le nom de domaine. L'idée derrière était également de mettre l'accent sur la pluralité des usages, il y a *des* publicodes (code publique), autant que de domaine représentés. Mais ce n'est pas très intuitif et apporte de la confusion. 

Peut-être demander à @laem ce qu'il en pense également ?

**Comment by laem** [comment]
Malheureusement je ne crois pas qu'il y ait de TLD .code 😄 
Personnellement, je ne vois pas trop de problèmes à jongler entre les deux. Perso j'utilise le pluriel (je crois ?). 

"Nos Gestes Climat est basé sur le langage publicodes"
"Nos Gestes Climat est basé sur le langage publicode"

Les deux me semblent compréhensibles.

**Comment by mquandalle** [comment]
Oui les deux sont clairement compréhensible en lecture, mais avoir les deux écritures qui co-existent peut poser des problèmes dans les noms de variables, etc. Par exemple faut il faire `npm install publicodes` ou `npm install publicode` ? J'avoue que même moi je ne sais pas sans aller vérifier dans la documentation !

**Comment by laem** [comment]
En tout cas, le domaine est renouvelé bientôt :D

**Comment by mquandalle** [comment]
Tu veux ma CB ? 😁

**Comment by laem** [comment]
C'est Ecolab qui paie ;-)

**Comment by johangirod** [comment]
On dirait que le `s` a été finalement choisi : 
- Pour les noms de package
- Dans notre base code, `publicodes` renvoi à 38 résultats, `publicode`, 8
- Le `s` permet de mettre en lumière la pluralité des règles, des algorithme public, des usages.

Je pense qu'il faudrait : 
- [x] Supprimer les dernières références à `publicode` 
- [ ] Ajouter un petit paragraphe dans la documentation "Pourquoi `publicodes` est au pluriel"
